test/aws-config/mzcompose.py

```diff
@@ -0,0 +1,251 @@
+            f"--var=denied-role-arn={denied.arn}",
+        ]
+
+        c.up(MZ.name)
+        write_aws_config(
+            c,
+            profile_contents,
+            MZ.name,
+        )
+        c.wait_for_materialized(MZ.name)
+        c.run(TESTDRIVE.name, *td_args, "test.td")
+
+        # Historically, a missing aws config file would cause all SQL
+        # commands to hang entirely after a restart, this no longer happens
+        # but this step restarts to catch it if it comes back.
+        c.stop(MZ.name)
+        c.rm(MZ.name)
+
+        c.up(MZ.name)
+
+        if args.restart_sleep_seconds > 0:
+            print(
+                "Sleeping for {}s to give materialized a chance to choke on permissions loading"
+            )
+            time.sleep(args.restart_sleep_seconds)
+
+        proc = c.exec(
+            MZ.name,
+            "sh",
+            "-c",
+            "mkdir -p ~/.aws; touch ~/.aws/config; cat ~/.aws/config",
+            capture=True,
+        )
+        assert not proc.stdout, "AWS config must NOT be present for restart tests"
+
+        c.run(
+            TESTDRIVE.name,
+            "--no-reset",
+            "test-restart-no-creds.td",
+        )
+
+        # now test that with added credentials things can be done
+        write_aws_config(
+            c,
+            profile_contents,
+            MZ.name,
+        )
+        c.run(TESTDRIVE.name, *td_args, "test-restart-with-creds.td")
+    finally:
+        errored = False
+        for role in created_roles:
+            try:
+                iam.delete_role_policy(RoleName=role.name, PolicyName=role.policy_name)
+            except Exception as e:
+                errored = True
+                print(
+                    f"> Unable to delete role policy {role.name}/{role.policy_name}: {e}"
+                )
+
+            try:
+                iam.delete_role(RoleName=role.name)
+                print(f"> Deleted IAM role {role.name}")
+            except Exception as e:
+                errored = True
+                print(f"> Unable to delete role {role.name}: {e}")
+
+        if errored:
+            raise UIError("Unable to completely clean up AWS resources")
+
+
+@dataclass
+class CreatedRole:
+    name: str
+    arn: str
+    policy_name: str
+
+
+def create_role(iam: IAMClient, effect: str, current_user: str) -> CreatedRole:
+    effect_name = effect.lower()
+    role_name = f"test-aws-config-{effect_name}-{SEED}"
+    policy_name = f"{effect_name}-{SEED}"
+
+    create_response = iam.create_role(
+        RoleName=role_name,
+        AssumeRolePolicyDocument=json.dumps(
+            {
+                "Version": "2012-10-17",
+                "Statement": {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": [
+                            current_user,
+                            "arn:aws:sts::834237029485:assumed-role/buildkite-15f2293-Role/*",
```

Comment:
Are you sure you don't want to point at a specific role here?

Maybe something like (I just made this up, get your correct role arn):
```
arn:aws:iam::834237029485:role/buildkite-15f2293-Role
```
Anything assuming that role will also be able to access it.

Reply:
Yeah at this point I'm just trying to figure out exactly what I'm supposed to type (that text is what I got from `aws sts get-caller-identity`, the `*` at the end is invalid), ideally this will be set in some more general way.